src/ui/components/shared/WorkspaceSettingsModal/OrganizationSettings.tsx

```diff
@@ -0,0 +1,172 @@
+import classNames from "classnames";
+import React, { useEffect, useRef, useState } from "react";
+import hooks from "ui/hooks";
+import { WorkspaceSettings } from "ui/types";
+
+const Label = ({ className, children }: { className?: string; children: React.ReactNode }) => {
+  return (
+    <div className={classNames(className, "w-5/12 flex-shrink-0")}>
+      <label>{children}</label>
+    </div>
+  );
+};
+
+const Input = ({ children }: { children: React.ReactNode }) => {
+  return <div className="w-8/12">{children}</div>;
+};
+
+const Row = ({ children }: { children: React.ReactNode }) => {
+  return <div className="flex items-center">{children}</div>;
+};
+
+const OrganizationSettings = ({ workspaceId }: { workspaceId: string }) => {
+  const { workspace } = hooks.useGetWorkspace(workspaceId!);
+  const updateWorkspaceSettings = hooks.useUpdateWorkspaceSettings();
+  const [message, setMessage] = useState(workspace?.settings?.motd || undefined);
+  const ref = useRef<NodeJS.Timeout | undefined>();
+
+  useEffect(() => {
+    if (ref.current) {
+      clearTimeout(ref.current);
+    }
+
+    ref.current = setTimeout(() => {
+      updateWorkspaceSettings({
+        variables: {
+          workspaceId,
+          motd: message,
+        },
+      });
+    }, 500);
+  }, [message]);
+
+  const updateFeature = (features: Partial<WorkspaceSettings["features"]>) => {
+    updateWorkspaceSettings({
+      variables: {
+        workspaceId,
+        features,
+      },
+    });
+  };
+
+  if (!workspace) {
+    return null;
+  }
+
+  const disabled = !workspace.isOrganization;
+
+  return (
+    <div className={classNames("space-y-4", { "text-gray-500": disabled })}>
+      {disabled ? (
+        <p>
+          These features are only available to teams on our Organization Plan. Want to upgrade? Get
+          in touch!
+        </p>
+      ) : null}
+      <div className="text-xs uppercase font-semibold">Recordings</div>
+      <Row>
+        <Label>Disable Public Recordings</Label>
+        <Input>
+          <label className="flex items-center" htmlFor="disable_public_recordings">
+            <input
+              className={classNames("rounded-sm ml-0 text-sm", {
+                "bg-toolbarBackground": disabled,
+                "border-gray-300": disabled,
+              })}
+              disabled={disabled}
+              type="checkbox"
+              id="disable_public_recordings"
+              name="disable_public_recordings"
+              onChange={e => updateFeature({ recording: { public: !e.currentTarget.checked } })}
```

Comment:
Isn't it supposed to be `e.target` here? Same question for all the other similar onChange handlers.

Reply:
It doesn't really matter. `currentTarget` ensures the node is *this* node and not a descendent that bubbled this event up. In this case, they're the same.